upgrade-notes/backend_20250502_145608.md

```diff
@@ -1,15 +1,31 @@
-- if you extend any shopsys controller action, copy the new `#[AccessControlRule]` attribute from the parent class
-- `MenuItemsGrantedRolesSubscriber::removeItemFromMenu()` method was removed without replacement
-- `Shopsys\FrontendApiBundle\Controller\CustomerUserController` was moved to `Admin` subfolder (`Shopsys\FrontendApiBundle\Controller\Admin\CustomerUserController`) and uses the `#[Route]` and `#[AccessControlRule]` attributes
-- if you have added any custom roles in your `App\Model\Security\Roles` class, check the [docs](https://docs.shopsys.com/en/17.0/cookbook/adding-a-new-administration-page/#access-control) for the new way of adding the roles
+#### implement attribute-based access control system for admin routes and RBAC system ([#3933](https://github.com/shopsys/shopsys/pull/3933)) and ([#4072](https://github.com/shopsys/shopsys/pull/4072))
+
+- **RBAC System Overhaul**: Complete refactoring of role-based access control system
+    - `App\Model\Security\Roles` class was **removed** - custom roles are now defined using `RoleProviderInterface` implementations
+    - roles are now managed through context-aware providers (AdminRoleProvider, CoreAdminRoleProvider, etc.)
+    - see the [RBAC documentation](https://docs.shopsys.com/en/17.0/introduction/role-based-access-control/#creating-custom-roles) for the new role definition approach
+
+- **Administrator Role Management**:
+    - administrators with `ROLE_ALL` or `ROLE_ALL_VIEW` are automatically migrated to use system-managed "Role Groups" during database migration
+    - administrators can no longer edit their own roles for security
+    - `AdministratorFormType`, `AdministratorRoleGroupFormType`, and `CustomerUserRoleGroupFormType` now use the new `RolesType` form with permission grid instead of simple select dropdowns
+    - `RolesType` form was completely rewritten and split into sections with permission checkboxes (VIEW, CREATE, EDIT, DELETE)
+    - system role constants (`ROLE_ALL`, `ROLE_ALL_VIEW`, `ROLE_ADMIN`, etc.) are now located in the `SystemRole` class instead of being scattered across different files
+    - you can simplify admin role permissions by setting `simple_permissions: true` in `shopsys_administration.yaml` configuration to show only VIEW and FULL permissions instead of granular ones
+
+- **Access Control Changes**:
+    - access control is now managed through PHP attributes on controller actions: `#[ForRole('ROLE_NAME')]`, `#[CanView]`, `#[CanEdit]`, etc.
+    - no longer necessary `MenuItemsGrantedRolesSetting` was removed, the menu items are now removed automatically based on the access control setting
+
+- **Development Changes**:
+    - if you have extended `tests-acceptance-before` phing target, be sure to add `warmup` target phing call before `test-db-dump` target call
```

Comment:
hm, I would say the warmup is no longer necessary, it was there because of the access_control_rules.json file